File: code/ch02/2.1.2.twoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        mapper = {}
        for i in range(n):
            if target - nums[i] in mapper:
                # 实际上这里返回的索引顺序是不重要的
                # 即返回 [i, mapper[target - nums[i]]] 也是正确的
                return [mapper[target - nums[i]], i]
            else:
                mapper[nums[i]] = i

        return []

File: code/ch02/2.1.2.twoSum.cpp
class Solution {
public:
  vector<int> twoSum(vector<int> &nums, int target) {

    unordered_map<int, int> mapper;

    for (int i = 0; i < nums.size(); i++) {
      if (mapper.find(target - nums[i]) != mapper.end())
        return {mapper[target - nums[i]], i};

      mapper[nums[i]] = i;
    }
    return {};
  }
};